Change default transform to accommodate "repeater" and "relationship" fields.

michaelcd opened the pull request:
#### Add support for "repeater" field
Currently the default transform will remove "repeater" field items before it makes it to the `rooftop` local. This occurs because repeaters don't have a `value` property, instead they have a `fields` property, which is an array of the repeater items.

Spike rooftop line that uses `value`:
```javascript
post.content[f.name] = f.value
```

JSON for repeater items:
```
{
  "name": "case_images",
  "label": "Case Images",
  "class": "repeater",
  "fields": [...]
}
```

#### Add support for "relationship" field
Currently the default transform will keep "relationship" field items, but not touch them otherwise. This PR will transform the JSON for relationship items to look like the structure for regular items. Ex: Advanced fields for nested relationship items can now be found in `post.content`

michaelcd commented:
@jescalan Have you had a chance to look at this?

jescalan commented:
I have not! Sorry about that. You can always assign me to something if you want me to check it out 😁

michaelcd commented:
I can't assign things under static-dev unfortunately

jescalan commented on lib/index.js:
```diff
@@ -100,17 +100,33 @@ function validate (opts = {}) {
+      })
+      delete item.advanced
+    }
+  }
```
This is only used once, any reason you made it a function?

michaelcd replied:
So it can be called recursively on line 113 to transform the JSON for nested relationship items:
`item.content[field.name].map(mapItem)`

jescalan replied:
Ah totally missed that. Makes sense!

jescalan commented:
Ok so this looks good, but I would like to have some test coverage for it. Any chance we'd be able to add this in?

michaelcd commented:
Sure! I'll work on a few tests and push up some commits when they're ready.

jescalan commented on test/index.js:
```diff
@@ -213,3 +213,47 @@ test.cb('accepts template object and generates html', (t) => {
+})
+
+// create a chain of relationship items on rooftop
+// write test of relationships
```
??

michaelcd replied:
Ignore that. :) removed

michaelcd commented:
I went ahead and updated some failing tests in this PR as well. Content changes on the carrotcreativedemo account caused them to fail, so I just updated the expected values.

jescalan commented:
Awesome, this is looking great. Thanks for adding those tests! It's so weird that travis still isn't taking env values 😕
